src/App.svelte

```diff
@@ -0,0 +1,72 @@
+<script>
+  import { db } from "./firebase.js";
+  export let name;
+  let arrList = [];
+  let newItemText = "";
+  let newItemQuantity;
+
+  db.collection("grocery").onSnapshot((snapData) => {
```

Comment:
Each time this component re-renders, a new db collection event is created. This should happen only once when the component mounts. You can use svete `onMount` lifecycle method.

Reply:
Could you please give me more details what that? I was also confused on that topic with React.
What does "re-render" exactly mean? When does a component "re-render"?
When does the component mount?
For example, what does happen when I add an item to the list? Does the component mount or re-render?